### PR TITLE
Add gcpsm provider

### PR DESCRIPTION
**Pull request by gabibeyer**
Fix merge conflicts for the GCP Secret Manager provider

### CODE REVIEW

File: pkg/provider/gcp/secretmanager/secretsmanager.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	}
+	err = sm.SecretManagerClient.Close()
+	if err != nil {
+		return nil, err
+	}
+	return []byte(string(result.Payload.Data)), nil
+}
+
+// GetSecretMap returns multiple k/v pairs from the provider.
+func (sm *GCP) GetSecretMap(ctx context.Context, ref esv1alpha1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
```
Comment (jonatasbaldin) [inline pkg/provider/gcp/secretmanager/secretsmanager.go:117]: Is the `GetSecretMap` not possible today with GCP SM or will this still be implemented?

**Review by jonatasbaldin — CHANGES_REQUESTED** [review]
Hi there, thanks for the awesome work ✨ 

On all the error checking, you could follow some good practices:
- Define the custom error messages as constants, like in `pkg/provider/aws/provider.go`
- Not call `log` since the error logging is already done on `externalsecret_controller.go`
- Always include the original `err` message on the `return` statement

Would be also nice to have a couple of unit tests for this, like on AWS and Vault.

Also left a comment on the code itself.

Again, thanks a bunch 💃

**Comment by gabibeyer** [comment]
Hey @jonatasbaldin Thank you so much for the input! I completely forgot to set this PR as a `draft` :woman_facepalming. I will implement the best practices for error checking, add unit tests and look into the `GetSecretsMap`. Thank you again for the input!

**Comment by jonatasbaldin** [comment]
No worries, thanks again for the PR ❤️

__________________________

Jonatas Baldin

Blog <https://deployeveryday.com/> | GitHub
<https://github.com/jonatasbaldin> | Twitter
<https://twitter.com/jonatasbaldin>

**Comment by knelasevero** [comment]
@moolen @jonatasbaldin @mcavoyk any of you have any idea why check-diff is failing here? Seems to be fine locally :thinking:

**Comment by mcavoyk** [comment]
> any of you have any idea why check-diff is failing here? Seems to be fine locally

Running locally I see a diff in `go.mod`. I agree though that CI step is a bit of a black box, created small change to print diff on failure here https://github.com/external-secrets/external-secrets/pull/157

**Comment by knelasevero** [comment]
Thanks @mcavoyk :) 

@gabibeyer try rebasing and seeing the error on ci

**Comment by knelasevero** [comment]
Already tested this! :smile: Great work @gabibeyer @1aziz

**Pull request merged**